.github/workflows/run-unit-tests.yml

```diff
@@ -0,0 +1,26 @@
+name: Run unit tests
+
+on:
+  pull_request:
+    types: [opened,reopened,synchronize]
+
+jobs:
+  test-unit-jest:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup environment
+        uses: ./.github/actions/setup-environment
+
+      - name: test:coverage:jest:dev
+        run: yarn test:coverage:jest:dev
+
+      - name: test:coverage:jest
+        run: yarn test:coverage:jest
+
+      - uses: codecov/codecov-action@d9f34f8cd5cb3b3eb79b3e4b5dae3a16df499a70
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          fail_ci_if_error: true
```

Comment:
Looks like something here is using an action that uses an old version of Node.js. You'll need to update it.

> [test-unit-jest](https://github.com/MetaMask/metamask-extension/actions/runs/9780888786/job/27003615234)
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: codecov/codecov-action@d9f34f8cd5cb3b3eb79b3e4b5dae3a16df499a70. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.


https://github.com/MetaMask/metamask-extension/actions/runs/9780888786

Reply:
It is due to: codecov/codecov-action@d9f34f8cd5cb3b3eb79b3e4b5dae3a16df499a70. We need permission to update to v4.

Reply:
updated as permission was granted